### PR TITLE
mpdecimal/2.4.2: migrate to Conan v2

### PR DESCRIPTION
**Pull request by valgur**

### CODE REVIEW

**Comment by valgur** [comment]
Closing temporarily to avoid unnecessary load on the CI. Will reopen when I'm actively working on the PR again.

**Review by jwillikers — APPROVED** [review]

**Review by samuel-emrys — APPROVED** [review]

File: recipes/mpdecimal/2.4.2/conanfile.py
Comment (RubenRBS) [inline recipes/mpdecimal/2.4.2/conanfile.py:1]: Is 2.4.2 version really necessary? I'd be all in for removing this version and simplifying the recipes a lot

**Review by RubenRBS — COMMENTED** [review]
Thanks! Looks good, but I have a suggestion for removing 2.4.2, let me know what you think @valgur

File: recipes/mpdecimal/2.4.2/conanfile.py
Comment (samuel-emrys) [reply recipes/mpdecimal/2.4.2/conanfile.py:1]: The comment in the [original commit adding cpython](https://github.com/conan-io/conan-center-index/commit/e5e3def4ace88be9118b86245179b4d98180416f) that's relevant to this is:

> * cpython: newer mpdecimal is not compatible with older cpython releases

Which goes along with this code block:

https://github.com/conan-io/conan-center-index/blame/edf1c5a297df013a55cb2d6ac7bf284681a86f33/recipes/cpython/all/conanfile.py#L182-L185

So, to support these older python versions I'd suggest that yes it is necessary

Comment (valgur) [reply recipes/mpdecimal/2.4.2/conanfile.py:1]: @RubenRBS I would have favored dropping it as well, but it's still required for older Python versions. I don't see it causing any further maintenance overhead, so keeping it around should not be much of an iasue.

**Review by RubenRBS — APPROVED** [review]

**Pull request merged**